src/ray/core_worker/transport/raylet_transport.h

```diff
@@ -32,6 +33,12 @@ class CoreWorkerRayletTaskReceiver : public rpc::WorkerTaskHandler {
                         rpc::AssignTaskReply *reply,
                         rpc::SendReplyCallback send_reply_callback) override;
 
+  // TODO(ekl) this shouldn't be in the raylet task receiver
```

Comment:
Why does this need to be here?

Reply:
I moved it. Since the worker and direct actor services are separate, this means we now need two rpc clients to each worker, which is a bit weird but probably ok since clients are lightweight.